### DISCUSSION TITLE
Simplify usage of bare hostnames for local servers

### DISCUSSION THREAD

**Discussion by stueB** [arendst/Tasmota#20284] [Ideas]
Environment: 
My Tasmota clients switch between two different Wi-Fi networks (Ap1 and Ap2 are set) with their own respective domain due to different conditions.

Cause: 
It seems that these Tasmota clients, unlike Windows-, Linux-, and Android- clients that also switch between my Wi-Fis, do not adopt the "search domain" transmitted by the respective DHCP server.

Problem: 
As a result, pure hostnames such as "myHost" for MqttHost, Ntpserver1, dnsserver, etc. cannot be completed to the respective FQDN (myHost.mydomain1.lan or myHost.mydomain2.lan) by Tasmota, causing the mediation of the required IP address from the local DNS server to fail.

Question: 
Would it be possible, or is it planned to establish ipaddress6 (or another number), to manually specify a "search domain", or to automatically adopt it via DHCP (option/code 119)?

**Comment by s-hadinger** [comment]
No it is not planned, and in my experience this brings its own set of problems. I don't even think that LwIP supports search domains (or I didn't find it). This means that even if we try to implement it for Tasmota, it won't work for native protocols such as NTP.

An alternative would be to have a FQDN that resolves to different IP addresses depending on the location. This is typically how it is done in entreprise environments.
